### ISSUE TITLE
After 3.22 update, it seems geojson export does not include CRS  when different from CRS84

### ISSUE THREAD

**Issue by simogeo** [qgis/QGIS#46383]
### What is the bug or the crash?

Geojson export does not include anymore (after 3.22 update ?) CRS definition in property, except when CRS is WGS84 !

```json
"crs": { "type": "name", "properties": { "name": "urn:ogc:def:crs:OGC:1.3:CRS84" } },
```

Resulting layers are not suitable

### Steps to reproduce the issue

1. Import any layers 
2. Right click to export in Geojson format to CRS different from WGS84
3. Import resulting layer ... that does not work
4. open resulting geojson file in text editor  and observe there is no CRS definition : 

**EPSG 2154 (incorrect) :** 

```json
{
"type": "FeatureCollection",
"name": "test",
"features": [
{ "type": "Feature", "properties": { "ID": "SURFACTI0000000023224238", "NOM_CHF": "Mairie de Brest", "STATUT": "Sous-préfecture", "INSEE_COM": "29019" }, "geometry": { "type": "Point", "coordinates": [ 4230288.326097229495645, -331820.483097876247484 ] } },
{ "type": "Feature", "properties": { "ID": "SURFACTI0000000056486935", "NOM_CHF": "Mairie de Valenciennes", "STATUT": "Sous-préfecture", "INSEE_COM": "59606" }, "geometry": { "type": "Point", "coordinates": [ 4069581.320086148101836, 250631.436038092302624 ] } }
]
}
````

**EPSG 4326 (correct) :** 
```json
{
"type": "FeatureCollection",
"name": "chef-lieu-Brest-Valenciennes",
"crs": { "type": "name", "properties": { "name": "urn:ogc:def:crs:OGC:1.3:CRS84" } },
"features": [
{ "type": "Feature", "properties": { "ID": "SURFACTI0000000023224238", "NOM_CHF": "Mairie de Brest", "STATUT": "Sous-préfecture", "INSEE_COM": "29019" }, "geometry": { "type": "Point", "coordinates": [ -4.485052434155741, 48.391323734310731 ] } },
{ "type": "Feature", "properties": { "ID": "SURFACTI0000000056486935", "NOM_CHF": "Mairie de Valenciennes", "STATUT": "Sous-préfecture", "INSEE_COM": "59606" }, "geometry": { "type": "Point", "coordinates": [ 3.524197715226201, 50.357856468432551 ] } }
]
}

````

### Versions

Version de QGIS | 3.22.1-Białowieża | Révision du code | 663dcf8fb9
-- | -- | -- | --
Version de Qt | 5.12.8
Version de Python | 3.8.10
Version de GDAL/OGR | 3.0.4
Version de Proj | 6.3.1
Version de la base de données du registre EPSG | v9.8.6 (2020-01-22)
Compilé avec GEOS | 3.8.0-CAPI-1.13.1 | Utilisé avec GEOS | 3.8.0-CAPI-1.13.1
Version de SQLite | 3.31.1
Version de PDAL | 2.0.1
Version du client PostgreSQL | 12.9 (Ubuntu 12.9-0ubuntu0.20.04.1)
Version de SpatiaLite | 4.3.0a
Version de QWT | 6.1.4
Version de QScintilla2 | 2.11.2
Version de l'OS | Ubuntu 20.04.3 LTS
  |   |   |  
Extensions Python actives
nettoyeur | 3.3.0
mask | 1.10.1
Qgis2threejs | 2.6
qgis_resource_sharing | 0.16.0
vector_tiles_reader | 3.2.2
ORStools | 1.4.0
shapetools | 3.4.1
valhalla | 2.2.1
cartogram3 | 3.0.4
DataPlotly | 3.8.1
french_locator_filter | 1.0.0
SRTM-Downloader | 3.1.15
Beeline | 0.2.0
QuickOSM | 2.0.0
OSMtools | 3.2
QNEAT3 | 1.0.4
TerrainShading | 0.9.3
grassprovider | 2.12.99
db_manager | 0.1.20
MetaSearch | 0.3.5
sagaprovider | 2.12.99
processing | 2.12.99

</body></html>

### Supported QGIS version

- [X] I'm running a supported QGIS version according to the roadmap.

### New profile

- [ ] I tried with a new QGIS profile

**Comment by gioman** [comment]
@simogeo works fine here

```
{
"type": "FeatureCollection",
"name": "cccc",
"crs": { "type": "name", "properties": { "name": "urn:ogc:def:crs:EPSG::2154" } },
"features": [
{ "type": "Feature", "properties": { "gid": 14, "area_inun": "menos de 10 ha", "fim": "Sem fim determinado", "ai": 4.0, "name": 14 }, "geometry": { "type": "Point", "coordinates": [ 138074.005347719736164, 26704.322655655676499 ] } },
...
...
...
```

**Comment by simogeo** [comment]
@gioman : thanks, do you think, it can be a dependency issue ? or OS related issue ? I'm under Ubuntu ?

**Comment by gioman** [comment]
@simogeo still ok for me on Ubuntu 20.04.

**Comment by simogeo** [comment]
Thanks again. I will close the issue and will enquire on my side ...

**Comment by simogeo** [comment]
Actually, I re-installed qgis after removing it and I still experience the same issue. I will still investigate but I re-open it just in case anyone else has the same issue.

**Comment by gioman** [comment]
> I will still investigate but I re-open it just in case anyone else has the same issue.

@simogeo it seems that this could be a local issue, have you tried on a different machine?

**Comment by simogeo** [comment]
I have only one machine. But yes it seems to be a local issue and I'd be happy to understand why this happens.

Does the 'Export' feature relies on GDAL/OGR ?

**Comment by gioman** [comment]
> I have only one machine.

@simogeo you can try on a virtual machine.

**Comment by roya0045** [comment]
On windows the crs is included in 3.22.1

**Comment by simogeo** [comment]
It seems to work correctly with 3.22.2 1601ec46d0 version.
